lib/entry/FlavorDownloader.js

```diff
@@ -276,29 +263,43 @@ var FlavorDownloader = (function () {
             return Q.resolve();
         };
 
-        return Q.fcall(function () {
+        return Q.fcall(function() {
             return downloadFlavorManifest()
-                .then(function finishedDownload(playlistDestPath) {
+                // Finished downloading manifest
+                .then(function(playlistDestPath) {
                     return m3u8Handler.parseM3U8(playlistDestPath)
-                        .then(function (m3u) {
+                        .then(function(m3u) {
                             that.logger.debug("Received M3U8: \n%s", m3u);
                             return m3u;
-                        }).then(function (m3u) {
+                        })
+                        .then(function(m3u) {
                             return updateListOfNewChunksToProcess(m3u);
-                        }).then(function (newChunks) {
+                        })
+                        .then(function(newChunks) {
                             //generate updated m3u8
                             var m3u8generatorPromise = that.m3u8generator.update(newChunks);
                             // Download chunks - all the chunks that were pending download beforehand along with the new chunks
-                            var downloadChunksPromise = downloadChunks(that, baseUrl, Object.keys(that.chunksToDownload));
+                            var downloadChunksPromise = downloadChunks.call(that, baseUrl, Object.keys(that.chunksToDownload));
                             return Q.allSettled([downloadChunksPromise, m3u8generatorPromise]);
-                        }).then(function (results) {
+                        })
+                        .then(function(results) {
+                            // TODO: What if chunklist hasn't changed since last iteration? -> Gad
+                            var tmp = that.m3u8generator.getCurrentManifest();
+                            if (tmp.items.PlaylistItem.length > 0) {
+                                that.logger.info("Finished downloading new chunks, raising 'newTsFiles' event");
+                                that.emit('newTsFiles', tmp.items.PlaylistItem.length);
```

Comment:
please change the event to have in parameters the list of new chunks, deleted chunks and list of current chunks